sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/async_challenge_auth_policy.py

```diff
@@ -21,13 +24,16 @@ class AsyncChallengeAuthPolicy(ChallengeAuthPolicyBase, AsyncHTTPPolicy):
             )
             if request.http_request.body:
                 # no_body was created with request's headers -> if request has a body, no_body's content-length is wrong
+                _LOGGER.debug("no_body was created with request's headers."
+                              "If request has a body, no_body's content length is wrong")
                 no_body.headers["Content-Length"] = "0"
 
             challenger = await self.next.send(PipelineRequest(http_request=no_body, context=request.context))
             try:
                 challenge = self._update_challenge(request, challenger)
             except ValueError:
                 # didn't receive the expected challenge -> nothing more this policy can do
+                _LOGGER.debug("Did not receive the expected challenge.")
```

Comment:
>debug [](start = 24, length = 5)

Should this be an error or warning rather than debug?

Reply:
I just talked to Johan and based on error and warning both being of a higher priority and based on what the other services are doing, debug is the best option

Reply:
Should it be logged at all? Unless the client's request minus body and auth token was acceptable to Key Vault, we're about to return an error response to the client, which will then raise an exception.